packages/core/src/index.js.flow

```diff
@@ -75,7 +75,9 @@ declare export function switchLatest <A> (s: Stream<Stream<A>>): Stream<A>
 
 declare export function merge <A, B> (s1: Stream<A>, s2: Stream<B>): Stream<A | B>
 declare export function merge <A, B> (s1: Stream<A>): (s2: Stream<B>) => Stream<A | B>
-declare export function mergeArray <A> (ss: Array<Stream<A>>): Stream<A>
+
+type MergeArray = <A>(Array<Stream<A>>) => Stream<A>
+declare export function mergeArray <S> (ss: S): $Call<MergeArray, S>
```

Comment:
@unscriptable This isn't quite the same as the ZipArray case (which I still don't know how to do in Flow), but it's a useful technique nonetheless.  It basically turns an array type into a union of the array's value types (removing the extra Stream nesting in this case).